pkg/server/plugin/nodeattestor/gcp/iit.go

```diff
@@ -17,74 +19,80 @@ import (
 const (
 	tokenAudience = "spire-gcp-node-attestor"
 	googleCertURL = "https://www.googleapis.com/oauth2/v1/certs"
+	svidPrefix    = "spiffe://{{ .TrustDomain }}/spire/agent"
```

Comment:
I went back and forth on hardcoding this prefix (which is a bit awkward because TrustDomain needs to be templated in) vs. validating the config option. I decided to go this route which makes the _defaultAgentSVIDTemplate statement a real mouthful but it seemed better than needing all users to configure the prefix correctly and then validate it in Configure (do we care about spaces in `{{ .TrustDomain }}`? seemed like a nightmare to validate).

Reply:
seems like the SVID var and const might be better in a common location, since aws will want the same

Reply:
Agreed

Reply:
hardcoding the prefix is a good idea because I think there are some assumptions built in and things might get wacky if the prefix isn't what we expect

idutil package in common directory might be a good place for this stuff. could use server there too e.g. `ServerIDPath` and `AgentIDPathPrefix`.

Could also expose some helper functions there like `func ServerID(trustDomain string) url.URL` and `func AgentID(trustDomain, suffix string) url.URL`. Something like this would definitely be useful elsewhere in the codebase... I generally like keeping SPIFFE IDs as URL types wherever I can because it helps to make sure things are correctly formatted... my two cents at least